clients/horizonclient/error_helpers.go

```diff
@@ -1,26 +1,32 @@
 package horizonclient
 
-import "github.com/stellar/go/support/errors"
+import (
+	"net/http"
+
+	"github.com/stellar/go/support/errors"
+)
 
 // IsNotFoundError returns true if the error is a horizonclient.Error with
 // a not_found problem indicating that the resource is not found on
 // Horizon.
 func IsNotFoundError(err error) bool {
-	var hErr *Error
-
-	err = errors.Cause(err)
-	switch err := err.(type) {
-	case *Error:
-		hErr = err
-	case Error:
-		hErr = &err
+	hErr := GetError(err)
+	if hErr == nil {
+		return false
 	}
 
+	return hErr.Problem.Type == "https://stellar.org/horizon-errors/not_found"
+}
+
+// IsHorizonAPITimeoutError returns true if the error is a horizonclient.Error with
+// a timeout problem indicating that Horizon timed out.
```

Comment:
Can we add some more commentary here outlining the things we've discussed in this PR? I don't want anyone to rely on this in ways they shouldn't.

```suggestion
// IsHorizonAPITimeoutError returns true if the error is a horizonclient.Error 
// with a timeout Problem. This indicates that Horizon timed out in a 
// well-defined, intentional way:
//     https://stellar.org/horizon-errors/not_found
// but is not a catch-all method for detecting other timeout errors.
```